File: js/src/types/attestation-model.ts
import type { AttestationEvidence } from './attestation-common';
import type { ChutesModelAttestation } from './attestation-chutes';

/** Raw evidence from NEAR AI's own model-serving TEE fleet. */
export type NearModelAttestation = AttestationEvidence & {
  readonly provider: 'near';
  /** Quote-bound model public key used by the selected E2EE protocol, when provided. */
  signingPublicKey?: string;
  nvidiaPayload?: string;
  reportedQuoteData?: string;
};

/** Provider-specific model-serving TEE evidence returned by NEAR AI Cloud. */
export type ModelAttestation = NearModelAttestation | ChutesModelAttestation;

File: py/src/verifiable_ai_sdk/types/attestation_model.py
"""Raw model attestation evidence returned by NEAR AI Cloud."""

from __future__ import annotations

from dataclasses import dataclass

from .attestation_common import AttestationEvidence, SigningIdentity


@dataclass(frozen=True, kw_only=True)
class ModelAttestation(AttestationEvidence):
    """A model-serving TEE attestation."""

    reported_quote_data: str | None = None
    nvidia_payload: str | None = None


__all__ = ['ModelAttestation', 'SigningIdentity']
